File: eva-core/eva-core-cache/src/main/java/io/nerv/common/cache/config/CacheConfiguration.java
package io.nerv.common.cache.config;

import lombok.extern.slf4j.Slf4j;
import org.springframework.cache.annotation.CachingConfigurerSupport;
import org.springframework.cache.interceptor.KeyGenerator;
import org.springframework.context.annotation.Configuration;

@Slf4j
@Configuration
public class CacheConfiguration extends CachingConfigurerSupport {
    /*
     * 定义缓存数据 key 生成策略的bean 包名+类名+方法名+所有参数
     */
    public KeyGenerator keyGenerator() {
        return (o, method, objects) -> {
            //格式化缓存key字符串
            StringBuilder sb = new StringBuilder();
            //追加类名
            sb.append(o.getClass().getName());
            //追加方法名
            sb.append(method.getName());
            //遍历参数并且追加
            for (Object obj : objects) {
                sb.append(obj.toString());
            }
            log.debug("调用缓存Key : " + sb.toString());
            return sb.toString();
        };
    }
}

File: eva-core/src/main/kotlin/io/nerv/core/cache/config/CacheConfiguration.kt
package io.nerv.core.cache.config

import org.slf4j.LoggerFactory
import org.springframework.cache.annotation.CachingConfigurerSupport
import org.springframework.cache.interceptor.KeyGenerator
import org.springframework.context.annotation.Configuration
import java.lang.reflect.Method

@Configuration
open class CacheConfiguration : CachingConfigurerSupport() {
    val log = LoggerFactory.getLogger(this.javaClass)
    /*
     * 定义缓存数据 key 生成策略的bean 包名+类名+方法名+所有参数
     */
    override fun keyGenerator(): KeyGenerator {
        return KeyGenerator { o: Any, method: Method, objects: Array<Any> ->
            //格式化缓存key字符串
            val sb = StringBuilder()
            //追加类名
            sb.append(o.javaClass.name)
            //追加方法名
            sb.append(method.name)
            //遍历参数并且追加
            for (obj in objects) {
                sb.append(obj.toString())
            }
            log.debug("调用缓存Key : $sb")
            sb.toString()
        }
    }
}